misk/src/main/kotlin/misk/cloud/gcp/GoogleCloudModule.kt

```diff
@@ -0,0 +1,80 @@
+package misk.cloud.gcp
+
+import com.google.auth.Credentials
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.NoCredentials
+import com.google.cloud.datastore.Datastore
+import com.google.cloud.datastore.DatastoreOptions
+import com.google.cloud.http.HttpTransportOptions
+import com.google.cloud.storage.Storage
+import com.google.cloud.storage.StorageOptions
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.Service
+import com.google.inject.Provides
+import com.google.inject.Singleton
+import misk.cloud.gcp.datastore.DatastoreConfig
+import misk.cloud.gcp.storage.StorageConfig
+import misk.environment.Environment
+import misk.inject.KAbstractModule
+import misk.inject.addMultibinderBinding
+import misk.inject.to
+import misk.logging.getLogger
+import javax.inject.Inject
+
+/** Installs support for talking to real GCP services, either direct or via semulator */
+class GoogleCloudModule : KAbstractModule() {
+    override fun configure() {
+        binder().addMultibinderBinding<Service>().to<GoogleCloud>()
+    }
+
+    @Provides
+    @Singleton
+    fun provideServiceCredentials(env: Environment): Credentials =
+            if (env == Environment.DEVELOPMENT) NoCredentials.getInstance()
+            else ServiceAccountCredentials.getApplicationDefault()
```

Comment:
This is configurable in some way right? The default may not be the appropriate service account (unless I'm misunderstanding how this works)

Reply:
We always use the default service account for permissions when running in the actual GCP environment.

Reply:
What about the case when you have an app that isn't deployed to GCP, but
may still want to interact with GCD? Not an entirely uncommon setup, e.g.,
when building bridging apps.

Reply:
fair. can add more configurability down the line.